neuralmonkey/trainers/objective.py

```diff
@@ -0,0 +1,81 @@
+from abc import abstractproperty
+from typing import TypeVar, Union, Tuple, List, Optional, Generic
+import tensorflow as tf
+from typeguard import check_argument_types
+
+from neuralmonkey.decorators import tensor
+from neuralmonkey.model.model_part import GenericModelPart
+
+# pylint: disable=invalid-name
+ObjectiveWeight = Union[tf.Tensor, float, None]
+Gradients = List[Tuple[tf.Tensor, tf.Variable]]
+MP = TypeVar("MP", bound=GenericModelPart)
+# pylint: enable=invalid-name
+
+
+class Objective(Generic[MP]):
+    """The training objective.
+
+    Attributes:
```

Comment:
Nevim, jestli takhle ti atributi jsou v tom docstingu dobře. Nemělo by to spíš být na u každý tý metody?

Reply:
hotovo